### PR TITLE
Update to Spring Boot 3.3.0

### PR DESCRIPTION
**Pull request by reta**
Update to Spring Boot 3.3.0

### CODE REVIEW

File: benchmarks/pom.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
   <parent>
     <groupId>io.zipkin</groupId>
     <artifactId>zipkin-parent</artifactId>
-    <version>3.3.1-SNAPSHOT</version>
+    <version>3.4.0-SNAPSHOT</version>
```
Comment (codefromthecrypt) [inline benchmarks/pom.xml]: I don't think we should bump minor on this, rather bump minor afterwards when you add the elastic stuff.

File: docker/test-images/zipkin-mysql/Dockerfile [hunk 2]
```diff
@@ -28,7 +28,7 @@ HEALTHCHECK --interval=1s --start-period=30s --timeout=5s CMD ["docker-healthche
 ENTRYPOINT ["start-mysql"]
 
 # Use latest from https://pkgs.alpinelinux.org/packages?name=mysql (without the -r[0-9])
-ARG mysql_version=10.11.7
+ARG mysql_version=10.11.8-r0
```
Comment (codefromthecrypt) [inline docker/test-images/zipkin-mysql/Dockerfile]:
```suggestion
ARG mysql_version=10.11.8
```

**Review by codefromthecrypt — APPROVED** [review]
fine to make one for spring-boot only, but I tend to bump all deps before a patch release. If you are up to it, can you audit? Otherwise, I'll do post merge

File: benchmarks/pom.xml (hunk 1 above)
Comment (codefromthecrypt) [reply benchmarks/pom.xml]: so after we bump all the deps, we cut 3.3.1, and after that's out we bump your PR for 3.4.0

Comment (reta) [reply benchmarks/pom.xml]: Got it, sure, I will revert that

**Comment by reta** [comment]
> fine to make one for spring-boot only, but I tend to bump all deps before a patch release. If you are up to it, can you audit? Otherwise, I'll do post merge

Sure, will do!

**Comment by reta** [comment]
@codefromthecrypt we should be all set, thanks for looking!

**Comment by codefromthecrypt** [comment]
thanks a lot for the help, I'll cut a patch now!

**Comment by codefromthecrypt** [comment]
actually I will bump alpine first

**Pull request merged**